File: xiaomi/project/repository-of-wangke/MyApplication/app/src/main/java/com/example/myapplication/work0530/DetailActivity.java
package com.example.myapplication.work0530;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.Nullable;

import com.bumptech.glide.Glide;
import com.example.myapplication.R;

public class DetailActivity extends Activity {



    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.work0530_detail_main_activity);

        // 获取传递的数据
        Intent intent = getIntent();
        if (intent != null) {
            if (intent.hasExtra("imageUrl")) {
                // 如果是图片数据，设置图片到 ImageView
                String imageUrl = intent.getStringExtra("imageUrl");
                System.out.println(imageUrl);
                setImage(imageUrl);
            } else if (intent.hasExtra("text")) {
                // 如果是文本数据，设置文本到 TextView
                String text = intent.getStringExtra("text");
                System.out.println(text);
                setText(text);
            }
        }
    }

    private void setImage(String imageUrl) {
        ImageView imageView = findViewById(R.id.work0530_detail_image);
        int resId = Integer.parseInt(imageUrl);
        Glide.with(this).load(resId).into(imageView);
    }

    private void setText(String text) {
        TextView textView = findViewById(R.id.work0530_detail_text);
        textView.setText(text);
    }


}

File: xiaomi/project/repository-of-wangke/work0604/KotlinProject/app/src/main/java/com/example/myKotlinProject/DetailActivity.kt
package com.example.myKotlinProject

import android.os.Bundle
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.example.mykotlinlessson.R
import org.greenrobot.eventbus.EventBus

class DetailActivity : AppCompatActivity() {

    companion object {
        const val EXTRA_POSITION = "extra_position"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)

        val position = intent.getIntExtra(EXTRA_POSITION, -1)

        val textView = findViewById<TextView>(R.id.textView)
        val imageView = findViewById<ImageView>(R.id.image)
        val likeView = findViewById<TextView>(R.id.likeView)

        when (val item = ItemDataManager.items[position]) {
            is Item.TextItem -> {
                textView.let {
                    it.text = item.text
                    it.visibility = View.VISIBLE
                    imageView.visibility=View.GONE
                }
                likeView.run{
                    isSelected = item.like
                    setOnClickListener {
                        item.like = !item.like
                        isSelected = item.like
                    }
                }
            }
            is Item.ImageItem -> {
                imageView.let {
                    it.setImageResource(item.imageResId)
                    it.visibility=View.VISIBLE
                    textView.visibility=View.GONE
                }
                likeView.run{
                    isSelected = item.like
                    setOnClickListener {
                        item.like = !item.like
                        isSelected = item.like
                    }
                }
            }
        }
    }

}